### PR TITLE
Update setup script to deploy the test-resources ARM template instead…

### PR DESCRIPTION
**Pull request by azabbasi**
… of directly using CLI commands

### CODE REVIEW

File: sdk/iot/Azure.Iot.Hub.Service/tests/prerequisites/setup.ps1 [hunk 1]
```diff
@@ -69,27 +69,41 @@ if (-not $appId)
...
-    Write-Host "`nCreating IoT Hub $IotHubName`n"
-    az iot hub create -n $IotHubName -g $ResourceGroup --location $Region --output none --only-show-errors
+    throw "`nARM template was not found. Please make sure you have an ARM template file named test-resources.json in the root of the service directory`n"
 }
-$iotHubHostName = az iot hub show -n $IotHubName -g $ResourceGroup --query 'properties.hostName' --output tsv --only-show-errors
+
+# Deploy test-resources.json ARM template.
+az deployment group create --resource-group $ResourceGroup --name $IotHubName --template-file $armTemplateFile --parameters `
+    baseName=$IotHubName `
+    testApplicationOid=$applicationOId
```
Comment (azabbasi) [inline sdk/iot/Azure.Iot.Hub.Service/tests/prerequisites/setup.ps1:101]: I am adding Region to this parameters list

**Comment by vinagesh** [comment]
Sorry to bother again but commit message update please :)

File: sdk/iot/Azure.Iot.Hub.Service/tests/prerequisites/setup.ps1 [hunk 2]
```diff
@@ -100,21 +114,14 @@ Write-Host("Writing user config file - $fileName`n")
 $appSecretJsonEscaped = ConvertTo-Json $appSecret
 $config = @"
 {
-    "IotHubHostName": "https://$($iotHubHostName)",
+    "IotHubConnectionString": "$iotHubConnectionString",
+    "IotHubHostName": "$($iotHubHostName)",
```
Comment (vinagesh) [inline sdk/iot/Azure.Iot.Hub.Service/tests/prerequisites/setup.ps1:118]: not just "$iotHubHostName" ?

File: sdk/iot/Azure.Iot.Hub.Service/tests/prerequisites/setup.ps1 [hunk 3]
```diff
@@ -69,27 +69,41 @@ if (-not $appId)
...
 {
     Write-Host "`nCreating Resource Group $ResourceGroup in $Region`n"
     az group create --name $ResourceGroup --location $Region --output none
 }
 
-$hubExists = az iot hub list -g $ResourceGroup --query "[?name=='$IotHubName']" --output tsv --only-show-errors
-if (-not $hubExists)
+Write-Host "`nDeploying resources to $ResourceGroup in $Region`n"
+
+$armTemplateFile = "../../../test-resources.json";
```
Comment (vinagesh) [inline sdk/iot/Azure.Iot.Hub.Service/tests/prerequisites/setup.ps1:91]: Same as the other script. We should update to join with PsScript so you can run it from any directory.

**Review by vinagesh — APPROVED** [review]

**Pull request merged**